### PR TITLE
interaction of cmt_insert_file_header and pp_indent_in_guard with --frag

### PR DESCRIPTION
**Pull request by rickyelopez**
Addresses #3527

### CODE REVIEW

File: src/indent.cpp [hunk 1]
```diff
@@ -4527,7 +4527,8 @@ bool ifdef_over_whole_file(void)
    LOG_FUNC_ENTRY();
 
    // if requested, treat an #if that guards the entire file the same as any other #if
-   if (options::pp_indent_in_guard())
+   // if running as frag, assume #if is not a guard
+   if (options::pp_indent_in_guard() || cpd.frag)
```
Comment (gmaurel) [inline src/indent.cpp:4531]: Please use:
```.cpp
if (  options::pp_indent_in_guard() 
   || cpd.frag)
```

File: src/uncrustify.cpp [hunk 2]
```diff
@@ -1648,7 +1648,8 @@ static void do_source_file(const char *filename_in,
 
 static void add_file_header()
 {
-   if (!chunk_is_comment(Chunk::GetHead()))
+   // don't add the file header if running as frag
+   if (!chunk_is_comment(Chunk::GetHead()) && !cpd.frag)
```
Comment (gmaurel) [inline src/uncrustify.cpp:1652]: Here also the style:
```.cpp
if (  !chunk_is_comment(Chunk::GetHead())
   && !cpd.frag)
```

**Comment by rickyelopez** [comment]
Updated!

**Comment by gmaurel** [comment]
Instead of Addresses https://github.com/uncrustify/uncrustify/issues/3527
please use
fixes https://github.com/uncrustify/uncrustify/issues/3527

My merge will closes the Issue

**Comment by rickyelopez** [comment]
Done. Thanks for merging!

**Pull request merged**